runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -1101,6 +1101,29 @@ inlineDoubleMin(TR::Node *node, TR::CodeGenerator *cg)
    return doubleMaxMinHelper(node, cg, false);
    }
 
+extern TR::Register * 
+inlineMathFma(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR_ASSERT_FATAL(node->getNumChildren() == 3, 
+   "In function inlineMathFma, the node at address %p should have exactly 3 children, but got %u instead", node, node->getNumChildren());
+
+   TR::Register      * targetRegister      = cg->allocateRegister(TR_VRF);  
```

Comment:
I don't think the result should be stored in a vector register. The data type of `node` is a double or a float, so the result should live in an FPR. I think we could urn into a scenario here where the register allocator chooses to allocate VRF26 for example, then if the result of the FMA call gets used again for example to feed into an argument to another call we would fail.

I think you should be able to construct a test case for this as well. What happens if the result of one FMA call is used as an argument to another method?

Reply:
I tested the situation where the result of FMA call is used as an argument to 1. another FMA call 2. math.sqrt(). Both cases compiles and runs with no problem. Should I still write a test case for this?
I have change the `targetRegister` to be a `TR_FPR` in ebcff19.